sklearn/metrics/scorer.py

```diff
@@ -282,6 +282,91 @@ def check_scoring(estimator, scoring=None, allow_none=False):
             "have a 'score' method. The estimator %r does not." % estimator)
 
 
+def _check_multimetric_scoring(estimator, scoring=None, allow_none=False):
+    """Check the scoring parameter in cases when multiple metrics are allowed
+
+    Parameters
+    ----------
+
+    estimator : sklearn estimator instance
+        The estimator for which the scoring will be applied.
+
+    scoring : string, callable or None, optional, default: None
+        A single string (see :ref:`_scoring_parameter`) or a callable
+        (see :ref:`_scoring`) to evaluate the predictions on the test set.
+
+        For evaluating multiple metrics either give a list of unique strings
+        or a dict with names as keys and callables as values.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Single scorers returning a list/array of values may be wrapped
+        into multiple scorers that return one value each.
+
+        If ``None``, the estimator's default scorer, is used.
+        The return value in that case will be ``{'score': <default_scorer>}``.
+        If the estimator's default scorer is not available, a ``TypeError``
+        is raised.
+
+    allow_none : boolean, default: False
```

Comment:
If you want scorer to be None when doing `_check_multimetric_scoring(..., scorer)`... We don't check at grid search if the `scorer` is `None`, single-metric or multi-metric... We do it only at `_check_multimetric_scoring`...